typescript/web/src/components/labeling-tool/options-tool-bar/edit-selection-mode/toggle-button-group.tsx

```diff
@@ -0,0 +1,50 @@
+import { ButtonGroup, ButtonGroupProps, useRadioGroup } from "@chakra-ui/react";
+import * as React from "react";
+import { ToggleButtonProps } from "./toggle-button";
+
+interface ToggleButtonGroupProps<T> extends Omit<ButtonGroupProps, "onChange"> {
+  name?: string;
+  value: T;
+  defaultValue?: string;
+  onChange?: (value: T) => void;
+}
+
+export const ToggleButtonGroup = <T extends string>(
+  props: ToggleButtonGroupProps<T>
+) => {
+  const { children, name, defaultValue, value, onChange, isDisabled, ...rest } =
+    props;
+  const { getRootProps, getRadioProps } = useRadioGroup({
+    name,
+    defaultValue,
+    value,
+    onChange,
+  });
+
+  const buttons = React.useMemo(
+    () =>
+      React.Children.toArray(children)
+        .filter<React.ReactElement<ToggleButtonProps>>(React.isValidElement)
+        .map((button, index, array) => {
+          const isFirstItem = index === 0;
+          const isLastItem = array.length === index + 1;
+
+          const styleProps = {
+            ...(isFirstItem && !isLastItem ? { borderRightRadius: 0 } : {}),
+            ...(!isFirstItem && isLastItem ? { borderLeftRadius: 0 } : {}),
+            ...(!isFirstItem && !isLastItem ? { borderRadius: 0 } : {}),
+            ...(!isLastItem ? { mr: "-px" } : {}),
```

Comment:
is `"-px"` something valid?

Reply:
Apart from that, FYI, passing `undefined` in a style value is the equivalent of passing nothing. You can do things like:
```
const style = {
  borderRightRadius: cond1 ? 0 : undefined,
  borderLeftRadius: cond2 ? 0 : undefined,
  // ... 
};
```
I tend to prefer this syntax as you can easily see which keys are set. However, your way works better when there are multiple attributes to set for the same condition. 

In the end, it boils down to the same result so feel free to keep it like it is ;)